lib/rust/ensogl/core/src/profiler.rs

```diff
@@ -0,0 +1,28 @@
+//! Enso specific metadata logging utilities for use with the profiling framework.
+use enso_profiler;
+use serde;
+use serde::Serializer;
+use std::fmt::Display;
+use std::fmt::Formatter;
+
+
+/// Metadata that is logged within the Enso core libraries.
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, Eq, PartialEq)]
+pub enum Metadata {
+    /// An RPC event that was received from the backend.
+    RpcEvent(String),
+}
+
+impl Display for Metadata {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Metadata::RpcEvent(name) => f.collect_str(name),
+        }
+    }
+}
+
+/// Log an RPC Event to the profiling framework.
+pub fn log_rpc_event(event_name: &'static str) {
+    let event_logger = enso_profiler::MetadataLogger::new("RpcEvent");
```

Comment:
Each `MetadataLogger` registers itself separately with the logging framework. For performance we should create one per type, not one per event. It can be done with `lazy_static`.

Reply:
Good point. `lazy_static` seems not to be enough as `MetadataLogger` is not `Sync`, but `thread_local` does the trick.